src/Microsoft.Health.Dicom.Core/DicomCoreResource.resx

```diff
@@ -489,22 +489,49 @@ For details on valid range queries, please refer to Search Matching section in C
   <data name="UpsRsFeatureDisabled" xml:space="preserve">
     <value>UPS-RS feature is disabled.</value>
   </data>
-  <data name="InvalidProcedureStepState" xml:space="preserve">
-    <value>'{0}' is not a valid procedure step state for the specified WorkitemInstanceUID '{1}'. {2}</value>
-    <comment>{0} ProcedureStepState. {1} WorkitemInstanceUID. {2} Error code</comment>
+  <data name="InvalidProcedureStepStateTransition" xml:space="preserve">
+    <value>The workitem instance with UID: '{0}' can not transition to '{1}' state: {2}.</value>
+    <comment>{0} WorkitemInstanceUID. {1} ProcedureStepState. {2} Error code</comment>
   </data>
   <data name="MismatchSopInstanceWorkitemInstanceUid" xml:space="preserve">
     <value>SOPInstanceUID '{0}' in the payload does not match the workitem query parameter '{1}'.</value>
     <comment>{0} SOPInstanceUID. {1} WorkitemInstanceUID.</comment>
   </data>
+  <data name="MismatchAffectedSopInstanceWorkitemInstanceUid" xml:space="preserve">
+    <value>AffectedSOPInstanceUID '{0}' in the payload does not match the specified WorkitemInstanceUID '{1}'.</value>
+    <comment>{0} AffectedSOPInstanceUID. {1} WorkitemInstanceUID.</comment>
+  </data>
+  <data name="MissingWorkitemInstanceUid" xml:space="preserve">
+    <value>WorkitemInstanceUID is missing.</value>
+  </data>
   <data name="DuplicateTagValueNotSupported" xml:space="preserve">
     <value>Duplicate value '{0}' for tag '{1}' not supported in a sequence.</value>
     <comment>{0} String Value. {1} DicomTag.</comment>
   </data>
+  <data name="UnexpectedTag" xml:space="preserve">
+    <value>'{0}' DicomTag is not expected for '{1}' operation. [WorkitemInstanceUID '{2}']</value>
+    <comment>{0} Dicom Tag. {1} Action/Operation. {2} WorkitemInstanceUID.</comment>
+  </data>
   <data name="InvalidTransactionUID" xml:space="preserve">
     <value>Invalid TransactionUID for the specified WorkitemInstanceUID '{1}'.</value>
     <comment>{0} WorkitemInstanceUID.</comment>
   </data>
+  <data name="WorkitemInstanceNotFound" xml:space="preserve">
+    <value>The workitem instance with UID: '{0}' is not found.</value>
+    <comment>{0} Workitem UID.</comment>
+  </data>
+  <data name="WorkitemIsInFinalState" xml:space="preserve">
+    <value>The workitem instance with UID: '{0}' is already in the final state '{1}'. {2}.</value>
+    <comment>{0} Workitem UID. {1} Procedure Step State. {2} Error or Warning Code.</comment>
+  </data>
+  <data name="WorkitemUpdateIsNotAllowed" xml:space="preserve">
+    <value>The workitem instance with UID: '{0}' can not be updated currently.</value>
+    <comment>{0} Workitem UID.</comment>
+  </data>
+  <data name="WorkitemCancelRequestSuccess" xml:space="preserve">
```

Comment:
This message should probably align with the warning text in https://dicom.nema.org/medical/dicom/current/output/html/part18.html#sect_11.8.3.2

Reply:
Done.